src/Handler/DataDogUdpHandler.php

```diff
@@ -92,6 +94,21 @@ private function getSocket(): Socket
 
     protected function write(LogRecord $record): void
     {
+        if ($this->tagContext) {
+            $rootSpan = GlobalTracer::get()->getRootScope()?->getSpan();
+            if (! $rootSpan) {
+                throw new LogicException('No root span is active.');
+            }
+
+            foreach ($record->context as $key => $value) {
+                if (! is_scalar($value)) {
+                    continue;
+                }
+
+                $rootSpan->setTag($key, $value);
```

Comment:
Can you add a prefix here so we don't overlap with anything built-in?

Probably something like this:
```suggestion
                $rootSpan->setTag("monolog-context.$key", $value);
```